packages/editor-ui/src/views/CanvasAddButton.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+	<div :class="$style.container" :style="containerCssVars" ref="container">
+		<n8n-tooltip placement="top" :value="showTooltip" :disabled="isScrimActive" popper-class="trigger-placeholder-button-tooltip">
+			<button :class="$style.button">
+				<font-awesome-icon icon="plus" size="lg" />
+			</button>
+			<template #content>
+				<p v-text="$locale.baseText('nodeView.canvasAddButton.addATriggerNodeBeforeExecuting')" />
+			</template>
+		</n8n-tooltip>
+		<p :class="$style.label" v-text="$locale.baseText('nodeView.canvasAddButton.chooseATrigger')" />
+	</div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+	name: 'CanvasAddButton',
+	props: {
+		showTooltip: {
+			type: Boolean,
+		},
+	},
+	data() {
+		return {
+			leftPosition: 0,
+			topPosition: 0,
+		};
+	},
+	computed: {
+		containerCssVars(): Record<string, string> {
+			return {
+				'--trigger-placeholder-left-position': `${this.leftPosition}px`,
+				'--trigger-placeholder-top-position': `${this.topPosition}px`,
+			};
+		},
+		isScrimActive(): boolean {
+			return this.$store.getters['ui/showCreatorPanelScrim'];
+		},
+	},
+	mounted() {
+		this.centerContainer();
```

Comment:
could this simply be a static value at the node view level? like the start node is always in a static position? why do we always need to recenter it? What if there's already a node in that position? We have some logic here to avoid collisions that you could possible reuse https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/views/canvasHelpers.ts#L432

Reply:
To me, the current behavior where we enable the button and show an error tooltip that says "Can't be executed, no triggers.." seems alright since the user would have to deliberately disable the trigger, and this would give them a hint to enable it again. But I think it's more of a UX question, so I asked Max/David [in Linear.](https://linear.app/n8n/issue/N8N-4639)

Reply:
It's one of the acceptance criteria that this button should be centered on the canvas:
```
When canvas is first opened, position of “Choose a Trigger..” should always be vertically and horizontally centered in viewport.
```
Thanks for the hint, I've reworked it to use this method to control the position of the button